common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/PasskeyReplyChannel.kt

```diff
@@ -0,0 +1,226 @@
+        postErrorInternal(
+            ReplyMessage.Error(domExceptionMessage = errorMessage, type = requestType)
+        )
+    }
+
+    /**
+     * Posts an error message based on a thrown exception.
+     *
+     * Maps credential exceptions to appropriate DOMException types.
+     *
+     * @param throwable Exception to convert and send.
+     */
+    fun postError(throwable: Throwable) {
+        postErrorInternal(throwableToErrorMessage(throwable))
+    }
+
+    /**
+     * Internal method to send error messages and log them.
+     */
+    private fun postErrorInternal(errorMessage: ReplyMessage.Error) {
+        val methodTag = "$TAG:postError"
+        send(errorMessage)
+        Logger.error(methodTag, "RequestType: $requestType, failed with error: $errorMessage", null)
+    }
+
+    /**
+     * Maps credential exceptions to DOMException error messages.
+     *
+     * Conversion table:
+     * - Cancellation/No credential → NotAllowedError
+     * - Interruption → AbortError
+     * - Configuration → NotSupportedError
+     * - Unknown → UnknownError
+     *
+     * @param throwable Exception to map.
+     * @return Error message with appropriate DOMException details.
+     */
+    private fun throwableToErrorMessage(throwable: Throwable): ReplyMessage.Error {
+        val errorMessage = throwable.message ?: "Unknown error (empty message)"
+
+        val exceptionName = when (throwable) {
+            // Cancellation exceptions - User cancelled
+            is CreateCredentialCancellationException,
+            is GetCredentialCancellationException,
+            is NoCredentialException -> DOM_EXCEPTION_NOT_ALLOWED_ERROR
+
+            // Interruption exceptions - Operation aborted
+            is CreateCredentialInterruptedException,
+            is GetCredentialInterruptedException -> DOM_EXCEPTION_ABORT_ERROR
+
+            // Provider configuration exceptions - Not supported
+            is CreateCredentialProviderConfigurationException,
+            is GetCredentialProviderConfigurationException -> DOM_EXCEPTION_NOT_SUPPORTED_ERROR
+
+            // Unknown exceptions
+            is CreateCredentialUnknownException,
+            is GetCredentialUnknownException -> DOM_EXCEPTION_UNKNOWN_ERROR
+
+            // Default case for other exceptions
+            else -> DOM_EXCEPTION_NOT_ALLOWED_ERROR
+        }
+
+        return ReplyMessage.Error(
+            domExceptionMessage = errorMessage,
+            domExceptionName = exceptionName,
+            type = requestType
+        )
+    }
+
+    /**
+     * Sends a message to JavaScript via the reply proxy.
+     */
+    @SuppressLint("RequiresFeature", "Only called when feature is available")
+    private fun send(message: ReplyMessage) {
+        val methodTag = "$TAG:send"
+        try {
+            replyProxy.postMessage(message.toString())
+        } catch (t: Throwable) {
+            Logger.error(methodTag, "Reply message failed", t)
```

Comment:
Could we potentially get stuck if an exception was thrown here and ESTS UX was still waiting for a response from us? Not sure what the mitigation would be in this case if the actual communication channel wasn't working, but just curious what the user might experience.